src/test/groovy/com/hierynomus/smbj/connection/SMBSessionBuilderSpec.groovy

```diff
@@ -68,4 +71,48 @@ class SMBSessionBuilderSpec extends Specification {
     then:
     smb30xExpectedDecryptionKey == generatedKey.getEncoded()
   }
+
+  def "should able to generate signingKey with encryption supported for Smb30x"() {
+    given:
+    def smb30xExpectedSigningKey = ByteArrayUtils.parseHex("8f5a6907bce9ec89b8f89e560d4e2e18")
+    def connectionContext = Mock(ConnectionContext)
+    connectionContext.supportsEncryption() >> true
+    connectionContext.getCipherId() >> SMB3EncryptionCipher.AES_128_CCM
+    def connection = Mock(Connection)
+    connection.getConnectionContext() >> connectionContext
+    def response = Mock(SMB2SessionSetup)
+    response.getSessionFlags() >> Collections.emptySet()
+    def context = new SessionContext()
+    context.setSessionKey(new SecretKeySpec(sessionKey, HMAC_SHA256_ALGORITHM))
+
+    when:
+    def sessionBuilder = new SMBSessionBuilder(connection, config, null)
+    sessionBuilder.deriveKeys(response, SMB2Dialect.SMB_3_0, context)
+
+    then:
+    context.getSigningKey() != null
+    smb30xExpectedSigningKey == context.getSigningKey().encoded
```

Comment:
Maybe add a check that the other keys are not derived in case of no encryption

Reply:
Is there any thing needed to add on this PR? I think the new commit already check other keys as well.